pkg/messaging/pubsub/server.go

```diff
@@ -0,0 +1,65 @@
+package pubsub
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"log"
+	"strings"
+
+	"github.com/pkg/errors"
+	"go.nanomsg.org/mangos/v3/protocol"
+	"nodemon/pkg/entities"
+	"nodemon/pkg/messaging"
+
+	"go.nanomsg.org/mangos/v3/protocol/pub"
+	_ "go.nanomsg.org/mangos/v3/transport/all"
+)
+
+func StartPubSubMessagingServer(ctx context.Context, nanomsgURL string, alerts <-chan entities.Alert) (protocol.Socket, error) {
+	if len(nanomsgURL) == 0 || len(strings.Fields(nanomsgURL)) > 1 {
+		log.Printf("Invalid nanomsg IPC URL for pubsub server'%s'", nanomsgURL)
+		return nil, errors.New("invalid nanomsg IPC URL for pub sub socket")
+	}
+
+	socketPubSub, err := pub.NewSocket()
+	if err != nil {
+		log.Printf("Failed to get new pub socket: %v", err)
+		return nil, err
+	}
+
+	if err := socketPubSub.Listen(nanomsgURL); err != nil {
+		log.Printf("Failed to listen on pub socket: %v", err)
+		return nil, err
+	}
```

Comment:
Close socket in case of error (On my mind trick with defer would be useful here)

Reply:
It shouldn't be close here, because there is `defer socket.Close` which will close the socket by the end of the function's work